### PR TITLE
Using pathlib to get config file

### PR DESCRIPTION
**Pull request by Mianto**
#369  @dstansby  I have tried to move from os.path to pathlib, Please have a look at it.

### CODE REVIEW

**Comment by Mianto** [comment]
@dstansby I've only changed config file from os.path to pathlib. Do you intend to move the whole heliopy from os.path to pathlib?

**Comment by abotiamnot** [comment]
Looks great!

File: heliopy/data/test/2018-03-01_sunspot_data_yearly.csv [hunk 1]
```diff
@@ -0,0 +1,318 @@
+1700.5;   8.3; -1.0;    -1;1
```
Comment (dstansby) [inline heliopy/data/test/2018-03-01_sunspot_data_yearly.csv]: These files need to be removed from the PR.

File: heliopy/util/config.py [hunk 2]
```diff
@@ -22,15 +23,16 @@ def get_config_file():
...
     # Get user configuration location
-    home_dir = os.path.expanduser("~")
-    config_file_1 = os.path.join(home_dir, '.heliopy', config_filename)
+    home_dir = Path.home()
+    config_file_1 = home_dir / '.heliopy' / config_filename
 
-    module_dir = os.path.dirname(heliopy.__file__)
-    config_file_2 = os.path.join(module_dir, config_filename)
+    module_dir = Path(heliopy.__file__)
+    config_file_2 = module_dir / '..' / config_filename
```
Comment (dstansby) [inline heliopy/util/config.py]: Why is the '..' bit here? I think the config file lives inside the module_dir, so this isn't needed.

Comment (Mianto) [reply heliopy/util/config.py]: heliopy.__file__ locates to /usr/local/lib/python3.5/dist-packages/heliopy/__init__.py so we need to take back a folder to config file.

**Pull request merged**